### DISCUSSION TITLE
Undefined/Multiple definitions when building C++ User Module

### DISCUSSION THREAD

**Discussion by Tobinator32** [micropython/micropython#16279] [ESP32]
Partly borrowed from here:
https://github.com/orgs/micropython/discussions/16260
With ongoing investigation it seems to not be related to up-wrap. Every Module build which contains mangled c++ names failes to build. Tested old and new versions of Micropython and IDF. Makes no difference. Any ideas how to get around this issue? By this time i tried a lot and nothing seems to work.

**Comment by stinos** [comment]
> Every Module build which contains mangled c++ names failes to build

Can you elaborate how to reproduce? I tried a lot of things, but ended up having to be *very* specific: https://github.com/micropython/micropython/issues/16284
